app/concepts/todo/create.rb

```diff
@@ -1,50 +1,42 @@
 class Todo::Create < Trailblazer::Operation
+  include Model
+  model Todo, :create
+
+  contract do
+    property :title, validates: { presence: true }
+  end
 
-  # Logic for creating a TODO
-  #     if `todo_list` is specified
-  #         create the `todo`
-  #         associated with `todo_list`
-  #     else
-  #         create the `todo`
-  #         create a `user`
-  #         create a default `todo_list`
-  #         associate `todo_list` with `user`
-  #         associate `todo` with `todo_list`
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
+    validate(params[:todo]) do
+      update_todo_list(contract.model, params)
+      contract.save
+    end
+  end
 
-    # build the todo
-    todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
+  private
 
-    # find or create the parent list for the todo
-    if params[:todo_list_id].blank?
-      todo_list = TodoList.find_or_create_by(name: "Default To-do List")
+  def update_todo_list(todo, params)
+    todo.list = get_or_create_todo_list(params)
+    todo.list.user = get_or_create_user(params)
+    todo.list.save
+  end
 
-      # Find or create the owner of the todo_list:
-      ##### If current_user_id is present, fetch that user
-      ##### otherwise, create a new one
-      current_user =
-        if params[:current_user] == nil
-          if params[:current_user_id] == nil
-            User.create!(fullname: "Guest")
-          else
-            User.find(params[:current_user_id])
-          end
-        else
-          params[:current_user]
-        end
+  def get_or_create_todo_list(params)
+    todo_list =
+      if params[:todo_list_id].present?
+        TodoList.find(params[:todo_list_id])
+      else
+        TodoList.find_or_create_by(name: "Default To-do List")
+      end
+  end
 
-      # assign the todo_list to the user
-      todo_list.user = current_user
-      todo_list.save!
-    else # i.e. if params[:todo_list_id] is not nil
-      todo_list = TodoList.find(params[:todo_list_id])
-    end
+  def get_or_create_user(params)
+    return params[:current_user] if params[:current_user].present?
 
-    # associate the todo with the todo_list
-    todo.list = todo_list
-    todo.save!
+    if params[:current_user_id].nil?
+      User.create!(fullname: "Guest")
+    else
+      User.find(params[:current_user_id])
```

Comment:
Nice edits. What do you think of moving this out into its own User::Create operation?